kpi/deployment_backends/base_backend.py

```diff
@@ -224,3 +259,31 @@ def get_submission(self, pk, requesting_user_id,
         except IndexError:
             pass
         return None
+
+    def set_status(self, status):
+        self.save_to_db({'status': status})
+
+    def save_to_db(self, updates: dict):
+        """
+        Persist values from deployment data into the DB.
+        `updates` is a dictionary of properties to update.
+        E.g: `{"active": True, "status": "not-synced"}`
+        """
+        # Avoid circular imports
+        # use `self.asset.__class__` instead of `from kpi.models import Asset`
+        now = timezone.now()
+        with transaction.atomic():
+            self.asset.__class__.objects.select_for_update() \
+                .filter(id=self.asset.pk).update(
+                _deployment_data=ReplaceValues(
+                    '_deployment_data',
+                    updates=updates,
+                ),
+                date_modified=now,
+            )
+        self.store_data(updates)
```

Comment:
i'm worried that the fancy footwork we do here with `_deployment_data` might be in vain if a later call to `asset.save()` ends up rewriting the entire `_deployment_data` column. i wonder if we should:
* reread all of `_deployment_data` from the database here, instead of setting only what we know has changed—but that still leaves a potentially long lag time between here and `asset.save()`
* reread `_deployment_data` in `asset.save()` before calling the superclass—but that means an extra database read for every single write
* somehow make normal `asset.save()` calls omit `_deployment_data`—but it's going to be ugly because django has `update_fields` but nothing like `exclude_fields`

this also makes me wonder why we store the sync status on the `Asset` instead of the `AssetFile` :thinking: maybe `AssetFile` should link to `AssetVersion` to tell whether or not it belongs to a draft or a deployed version

Reply:
idea
```python
a = Asset.objects.first()
afnames = [f.name for f in a._meta.fields]
afnames.remove('_deployment_data')
```

Reply:
moved to #3057 notes

Reply:
As per discussed,  `_deployment_data` cannot be altered by calling `.save()`